packages/subgraph/src/v1-analytics/mapping.ts

```diff
@@ -161,11 +174,24 @@ export function handleTransactionPrepared(event: TransactionPrepared): void {
 
   transaction.save();
 
+  // Get asset balance
+  let assetBalanceId = transaction.sendingAssetId.toHex() + "-" + event.params.router.toHex();
+  let assetBalance = AssetBalance.load(assetBalanceId);
+  if (assetBalance == null) {
+    assetBalance = new AssetBalance(assetBalanceId);
+    assetBalance.assetId = transaction.sendingAssetId;
+    assetBalance.router = event.params.router.toHex();
+    assetBalance.amount = new BigInt(0);
+    assetBalance.supplied = new BigInt(0);
+    assetBalance.locked = new BigInt(0);
+  }
+
   // router is providing liquidity on receiver prepare
   if (chainId == transaction.receivingChainId) {
     let assetBalanceId = transaction.receivingAssetId.toHex() + "-" + event.params.router.toHex();
     let assetBalance = AssetBalance.load(assetBalanceId);
     assetBalance.amount = assetBalance.amount.minus(transaction.amount);
+    assetBalance.locked = assetBalance.locked.add(transaction.amount);
     assetBalance.save();
```

Comment:
we don't save unless chainId == transaction.receivingChainId ..?

Reply:
if this is the case, the whole "get asset balance" above should maybe be in this block

Reply:
ah wow actually this made me catch that i wasn't handling the difference in sending/receiving asset balances properly in the id. but yeah, we only save if its the receiving chain cus thats the only time we update on prepare